apps/server/src/modules/pseudonym/service/pseudonym.service.ts

```diff
@@ -48,6 +71,42 @@ export class PseudonymService {
 		return pseudonym;
 	}
 
+	public async deleteByUserId(userId: string): Promise<number> {
+		if (!userId) {
+			throw new InternalServerErrorException('User id is missing');
+		}
+
+		const [deletedPseudonyms, deletedExternalToolPseudonyms] = await Promise.all([
+			this.deletePseudonymByUserId(userId),
+			this.deleteExternalToolPseudonymByUserId(userId),
+		]);
+
+		return deletedPseudonyms + deletedExternalToolPseudonyms;
+	}
+
+	private async findPseudonymByUserId(userId: string): Promise<Pseudonym[]> {
+		const pseudonymPromise: Promise<Pseudonym[]> = this.pseudonymRepo.findPseudonymsByUserId(userId);
+		return pseudonymPromise;
+	}
+
+	private async findExternalToolPseudonymByUserId(userId: string): Promise<Pseudonym[]> {
```

Comment:
Please use plural form in this method's name (_findExternalToolPseudonym**s**ByUserId_).